src/js/actions/facetLayoutActions.ts

```diff
@@ -0,0 +1,14 @@
+import {createStandardAction} from 'typesafe-actions';
+import {FacetLayoutRecord} from '../store/factory/FacetLayout';
+import {assignId} from '../util/counter';
+import {State} from '../store';
+import {Dispatch} from 'redux';
+
+export function addFacetLayout (payload: FacetLayoutRecord) {
```

Comment:
re: this file, facetLayoutReducer, and facetLayout.ts, i think they're related enough to core layout functionality that i would consider just adding the facet actions / reducer cases / record definitions to the existing layout files so that it's easier to find everything.